### PR TITLE
fix(workflows-sdk): Fix StepFunction typings and custom step name

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Fix step function typings
- Fix step name when custom config is passed

### CODE REVIEW

File: packages/workflows-sdk/package.json [hunk 1]
```diff
@@ -40,6 +40,6 @@
     "build": "rimraf dist && tsc --build",
     "watch": "tsc --build --watch",
     "test": "jest --runInBand --bail --forceExit",
-    "test:run": "../../node_modules/.bin/ts-node ./src/utils/_test.ts"
+    "test:run": "../../node_modules/.bin/ts-node ./src/utils/_playground.ts"
```
Comment (olivermrbl) [inline packages/workflows-sdk/package.json:43]: question: intentional?

**Comment by adrien2p** [comment]
@olivermrbl should we merge?

**Review by olivermrbl — APPROVED** [review]

File: packages/workflows-sdk/package.json (hunk 1 above)
Comment (adrien2p) [reply packages/workflows-sdk/package.json:43]: yes _test does not exists and was replace bu this one

**Pull request merged**